cli/command/telemetry_docker.go

```diff
@@ -0,0 +1,121 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+const otelConfigFieldName = "otel"
```

Comment:
We should call this `otelContextFieldName` I think; I also would like to see this moved to the shared context implementation and used from there in Compose/Buildx, though I don't think it's a blocker here.

Reply:
Changed the name. This shouldn't need to be exposed because Compose and Buildx would end up using this code and would remove their own implementations of this so I don't think a shared exposed constant is needed.